Applications/DataExplorer/DataView/DiagramView/DetailWindow.cpp

```diff
@@ -124,9 +124,8 @@ void DetailWindow::resizeWindow()
 
 void DetailWindow::addList(DiagramList* list)
 {
-	GeoLib::Color* c = GeoLib::getRandomColor();
-	QColor colour((*c)[0], (*c)[1], (*c)[2]);
-	delete c;
+	GeoLib::Color const c = GeoLib::getRandomColor();
```

Comment:
Here is something about it: http://en.cppreference.com/w/cpp/language/copy_initialization
> * If T is a class type and the type of other is cv-unqualified version of T or a class derived from T, __the converting constructors of T are examined and the best match is selected by overload resolution.__ The constructor is then called to initialize the object. 
[...]

> Copy-initialization is less permissive than direct-initialization: explicit constructors are not converting constructors and are not considered for copy-initialization.